src/main/java/io/fabric8/maven/docker/util/DockerFileUtil.java

```diff
@@ -155,16 +155,25 @@ public static FixedStringSearchInterpolator createInterpolator(MojoParameters pa
      * @param argLines list of string arrays containing lines with words
      * @return map of parsed arguments
      */
-    protected static Map<String, String> extractArgsFromLines(List<String[]> argLines) {
+    static Map<String, String> extractArgsFromLines(List<String[]> argLines, Map<String, String> argsFromBuildConfig) {
         Map<String, String> result = new HashMap<>();
         for (String[] argLine : argLines) {
             if (argLine.length > 1) {
-                updateMapWithArgValue(result, argLine[1]);
+                updateMapWithArgValue(result, argsFromBuildConfig, argLine[1]);
             }
         }
         return result;
     }
 
+    static String resolveArgValueFromStrContainingArgKey(String argString, Map<String, String> args) {
+        if (argString.startsWith("$") && args.containsKey(argString.substring(1))) {
+            return args.get(argString.substring(1));
+        } else if (argString.startsWith("${") && argString.endsWith("}") && args.containsKey(argString.substring(2, argString.length() - 1))) {
```

Comment:
I think spaces are also allowed here like in `${ foobar }`. So for extracting the variable name it might be to user a regular expression. you then could also collapse the if statements into a single regular expression.

Reply:
I.e I think a REs like 

```
^\$([^{}]+)$

^\$\{\s*([^{}\s]+)\s*\}$
```

would do the trick (not verified though)

Reply:
I'm trying out an arg with spaces like this in this Dockerfile:
```Dockerfile
ARG busyboxVersion
FROM busybox:${ busyboxVersion }
```
When I trying to build it with docker, I'm getting an error:
```bash
docker : $ docker build -t test-dockerfile --build-arg busyboxVersion=latest .
Sending build context to Docker daemon 3.072 kB
Error response from daemon: Dockerfile parse error line 2: FROM requires either one or three arguments
```